app/src/format.py

```diff
@@ -87,42 +92,52 @@ def format_bem_subsections(
     chunks = [c.chunk for c in chunks_with_scores]
     context = split_into_subsections(chunks)
     citation_to_numbers = dereference_citations(context, raw_response)
-
     citations_html = ""
-    for citation, citation_number in citation_to_numbers.items():
-        _accordion_id += 1
-        chunk = citation.chunk
-        subsection = citation.subsection
-
-        formatted_subsection = to_html(replace_bem_with_link(subsection))
-        bem_url_for_page = get_bem_url(chunk.document.name)
-        if chunk.page_number:
-            bem_url_for_page += "#page=" + str(chunk.page_number)
-
-        citation_headings = "<p>" + " → ".join(chunk.headings) + "</p>" if chunk.headings else ""
-        citation_body = f'<div class="margin-left-2 border-left-1 border-base-lighter padding-left-2">{formatted_subsection}</div>'
-        citation_link = (
-            (f"<p><a href={bem_url_for_page!r}>Open document to page {chunk.page_number}</a></p>")
-            if chunk.page_number
-            else ""
-        )
-        citations_html += f"""
-        <div class="usa-accordion" id=accordion-{_accordion_id}>
-            <h4 class="usa-accordion__heading">
-                <button
-                    type="button"
-                    class="usa-accordion__button"
-                    aria-expanded="false"
-                    aria-controls="a-{_accordion_id}">
-                    {citation_number}. {chunk.document.name}
-                </button>
-            </h4>
-            <div id="a-{_accordion_id}" class="usa-accordion__content usa-prose" hidden>
-                {citation_headings}
-                {citation_body}
-                {citation_link}
-            </div>
-        </div>"""
+
+    citations_by_document = combine_citations_by_document(citation_to_numbers)
+    for document, chunk_list in citations_by_document.items():
+        for chunk in chunk_list:
```

Comment:
Rename `chunk` and `chunk_list` so that they better reflect the types of those variables.